src/EnergyPlus/DaylightingManager.cc

```diff
@@ -6217,17 +6214,9 @@ void DayltgInteriorIllum(EnergyPlusData &state,
     int ISky;   // Sky type index
     int ISky1;  // Sky type index values for averaging two sky types
     int ISky2;
-    auto &DFSUHR = state.dataDaylightingManager->DFSUHR;       // Sun daylight factor for bare/shaded window
-    auto &BFSUHR = state.dataDaylightingManager->BFSUHR;       // Sun background luminance factor for bare/shaded window
-    auto &SFSUHR = state.dataDaylightingManager->SFSUHR;       // Sun source luminance factor for bare/shaded window
-    auto &HorIllSky = state.dataDaylightingManager->HorIllSky; // Horizontal illuminance for different sky types
-    auto &SetPnt = state.dataDaylightingManager->SetPnt;       // Illuminance setpoint at reference points (lux)
-    auto &GLRNDX = state.dataDaylightingManager->GLRNDX;       // Glare index at reference point
-    auto &GLRNEW = state.dataDaylightingManager->GLRNEW;       // New glare index at reference point
-    auto &SFSKHR = state.dataDaylightingManager->SFSKHR; // Sky source luminance factor for sky type (second index), bare/shaded window (first index)
-    auto &DFSKHR = state.dataDaylightingManager->DFSKHR; // Sky daylight factor for sky type (second index), bare/shaded window (first index)
-    auto &BFSKHR =
-        state.dataDaylightingManager->BFSKHR; // Sky background luminance factor for sky type (second index), bare/shaded window (first index)
```

Comment:
All these were the same. I just replaced this var with the expanded version.

[src/EnergyPlus/DaylightingManager.cc:6220]:(style),[constVariable],Variable 'DFSUHR' can be declared as reference to const
[src/EnergyPlus/DaylightingManager.cc:6221]:(style),[constVariable],Variable 'BFSUHR' can be declared as reference to const
[src/EnergyPlus/DaylightingManager.cc:6222]:(style),[constVariable],Variable 'SFSUHR' can be declared as reference to const
[src/EnergyPlus/DaylightingManager.cc:6223]:(style),[constVariable],Variable 'HorIllSky' can be declared as reference to const
[src/EnergyPlus/DaylightingManager.cc:6227]:(style),[constVariable],Variable 'SFSKHR' can be declared as reference to const
[src/EnergyPlus/DaylightingManager.cc:6228]:(style),[constVariable],Variable 'DFSKHR' can be declared as reference to const
[src/EnergyPlus/DaylightingManager.cc:6229]:(style),[constVariable],Variable 'BFSKHR' can be declared as reference to const

Reply:
I think CppCheck is getting confused.  I don't know that it can tell that the parentheses operator is a non-const operation.  So it tries  to get you to declare them as const.  But in reality it seems like the compiler is opting for the non-const [`ObjexxFCL::Array1D::operator()`](https://github.com/NREL/EnergyPlus/blob/bde23f2559f1467a62790c452dd5d102f078936b/third_party/ObjexxFCL/src/ObjexxFCL/Array1.hh#L859).  The best way to solve this is to change them to std::vector, but of course that can be a substantial effort for some variables.